content/blog/2020/05/2020-05-07-docker-windows-agents.adoc

```diff
@@ -0,0 +1,121 @@
+---
+layout: post
+title: "Windows Docker Agent Images"
+tags:
+- announcement
+- docker
+- platform-sig
+author: slide_o_mix
+opengraph:
+  image: /images/docker/dockerJenkins_social.png
+---
+
+We would like to announce the availability of Windows agent images for Docker.
+
+image:/images/docker/dockerJenkins.png[Jenkins and Docker, role=center, float=right, height=224]
+
+== New images
+
+* link:https://hub.docker.com/r/jenkins/agent[jenkins/agent] is a basic agent which bundles the agent.jar for agent <= => master communication. This is most useful as a base image for other images.
+
+* link:https://hub.docker.com/r/jenkins/inbound-agent[jenkins/inbound-agent] is an agent that is based on the jenkins/agent image above. It provides a wrapper script written in PowerShell to help specify the parameters to agent.jar.
+
+* link:https://hub.docker.com/r/jenkins/ssh-agent[jenkins/ssh-agent] is an image which has OpenSSH installed and should be used with the plugin:ssh-slaves[SSH Build Agents Plugin].
+
+== Deeper Dive
+
+
+=== jenkins/agent
+
+This is a base image for Docker, which includes the JDK and the Jenkins agent executable (agent.jar). This executable is an instance of the Jenkins Remoting library. Multiple images are published for different versions of the JDK (8 or 11)
+as well as windowsservercore and nanoserver versions. This image is mainly useful as a base image for other images (`FROM jenkins/agent:jdk8-nanoserver-1809` in your Dockerfile). When using this agent, select `Launch agent by connecting it to the master` in the `Launch method` drop-down box. You will also want to specify the `Remote root directory` as a directory you prefer, or a good default to specify is `C:\Users\jenkins\Agent`.
+
+The agent image has a user `jenkins` which is used to run tasks on the Windows agent. The `jenkins` user is a member of the Administrators group inside the container.
+
+When the agent is created, it will show you the information needed to pass to agent.jar to run the agent and have it connect to the master. Here is an example of that command:
+
+```
+java -jar agent.jar -jnlpUrl https://JENKINS_URL/jenkins/computer/AGENTNAME/slave-agent.jnlp -secret SOMESECRETVALUE -workDir "C:\Users\jenkins\Agent"
```

Comment:
Does not make much sense to me. If you were running an inbound agent (as this CLI syntax implies), why would you not just use the `inbound-agent` image? The base image is useful because you can launch it in stdio mode from all sorts of tools or scripts.

Reply:
I haven't used the base image very much, so I don't know much about what it is used for. If you have additional text that you can suggest, I would be more than happy to incorporate it.

Reply:
https://github.com/jenkinsci/docker-agent#usage for example

Reply:
Ok, thanks for the clarification. I will incorporate this type of example.